pysat/utils/files.py

```diff
@@ -0,0 +1,679 @@
+                        else:
+                            out_dict['search_string'] += '*'
+                        break
+            else:
+                raise ValueError("Couldn't determine formatting width")
+
+    return out_dict
+
+
+def search_local_system_formatted_filename(data_path, search_str):
+    """
+    Parses format file string and returns string formatted for searching.
+
+    Parameters
+    ----------
+    data_path : string
+        Top level directory to search files for. This directory
+        is provided by pysat to the instrument_module.list_files
+        functions as data_path.
+    search_str : string
+        String used to search for local files. For example,
+        `cnofs_cindi_ivm_500ms_????????_v??.cdf` or `inst-name-*-v??.cdf`
+
+    Returns
+    -------
+    files : list
+        list of files matching the specified file format
+
+    Note
+    ----
+    The use of ?s (1 ? per character) rather than the full wildcard * provides
+    a more specific filename search string that limits the false positive rate.
+
+    """
+
+    # Perform local file search
+    abs_search_str = os.path.join(data_path, search_str)
+    files = glob.glob(abs_search_str)
+
+    # Remove the specified data_path portion
+    files = [sfile.split(data_path)[-1] for sfile in files]
+
+    # Return the desired filename information
+    return files
+
+
+def update_data_directory_structure(new_template, test_run=True,
+                                    full_breakdown=False,
+                                    remove_empty_dirs=False):
+    """Update pysat data directory structure to match supplied template
+
+    Translates all of pysat's managed science files to a new
+    directory structure. By default, pysat uses the template string stored in
+    pysat.params['directory_format'] to organize files. This method makes
+    it possible to transition an existing pysat installation so it works
+    with the supplied new template.
+
+    Parameters
+    ----------
+    new_template : str
+        New directory template string. The default value for pysat is
+         `os.path.join(('{platform}', '{name}', '{tag}', '{inst_id}'))`
+    test_run : bool
+        If True, a printout of all proposed changes will be made, but the
+        directory changes will not be enacted. (default=True)
+    full_breakdown : bool
+        If True, a full path for every file is printed to terminal.
+        (default=False)
+    remove_empty_dirs : bool
+        If True, all directories that had pysat.Instrument data moved
+        to another location and are now empty are deleted. Traverses
+        the directory chain up to the top-level directories in
+        `pysat.params['data_dirs']`. (default=False)
+
+    Note
+    ----
+    After updating the data directory structures users should nominally
+    assign `new_template` as the directory format via:
+
+        pysat.params['directory_format'] = new_template
```

Comment:
```suggestion
    assign `new_template` as the directory format via
    ::

        pysat.params['directory_format'] = new_template
```
Original: 
![Screen Shot 2021-02-23 at 7 08 49 PM](https://user-images.githubusercontent.com/19592220/108925457-a2528680-760a-11eb-9a2d-02a0083d5ad8.png)
Suggestion:
![Screen Shot 2021-02-23 at 7 12 08 PM](https://user-images.githubusercontent.com/19592220/108925675-11c87600-760b-11eb-98e5-125bed88e851.png)